### PR TITLE
Add clientName parameter to AssessmentService.findAssessmentByKey()

### PR DESCRIPTION
**Pull request by jeffjohnson9046**
Add `clientName` parameter to `AssessmentService.findAssessmentKey()` method call.

### CODE REVIEW

File: service/src/main/java/tds/exam/services/AssessmentService.java [hunk 1]
```diff
@@ -10,8 +10,10 @@
 public interface AssessmentService {
     /**
      * Finds the {@link tds.assessment.Assessment}
+     *
+     * @param clientName The name of the client (e.g. SBAC or SBAC_PT)
      * @param key unique key for the assessment
      * @return {@link tds.assessment.Assessment the assessment}
      */
-    Optional<Assessment> findAssessmentByKey(String key);
+    Optional<Assessment> findAssessmentByKey(String clientName, String key);
```
Comment (jeffjohnson9046) [inline service/src/main/java/tds/exam/services/AssessmentService.java]: Throughout most of our code, the signature for `findAssessmentByKey` is `findAssessmentByKey(clientName, assessmentKey`.  However, in the TDS_AssessmentService, the `AssessmentQueryRepository` has the signature set to `findAssessmentByKey(assessmentKey, clientName)` (the parameters are reversed).  At some point, we should probably come back and clean that up, but I think that's a separate pull request.

Comment (millerts) [inline service/src/main/java/tds/exam/services/AssessmentService.java]: I vote that you change it now.

Comment (jtreuting) [inline service/src/main/java/tds/exam/services/AssessmentService.java]: I know I'm totally late to the game here, but do we need the `ByKey` part?  Seems like that is the implied behavior of an `findAssessment` call.

Comment (millerts) [inline service/src/main/java/tds/exam/services/AssessmentService.java]: No.  I'll make the change in a PR I'll have up in a few

**Pull request merged**